samtranslator/model/apigatewayv2.py

```diff
@@ -57,39 +59,156 @@ class ApiGatewayV2ApiMapping(Resource):
 
 class ApiGatewayV2Authorizer(object):
     def __init__(
-        self, api_logical_id=None, name=None, authorization_scopes=[], jwt_configuration={}, id_source=None,
+        self,
+        api_logical_id=None,
+        name=None,
+        authorization_scopes=[],
+        jwt_configuration={},
+        id_source=None,
+        function_arn=None,
+        function_invoke_role=None,
+        identity=None,
+        authorizer_payload_format_version=None,
+        enable_simple_responses=None,
     ):
         """
         Creates an authorizer for use in V2 Http Apis
         """
-        if authorization_scopes is not None and not isinstance(authorization_scopes, list):
-            raise InvalidResourceException(api_logical_id, "AuthorizationScopes must be a list.")
-
-        # Currently only one type of auth
-        self.auth_type = "oauth2"
-
         self.api_logical_id = api_logical_id
         self.name = name
         self.authorization_scopes = authorization_scopes
-
-        # Validate necessary parameters exist
-        if not jwt_configuration:
-            raise InvalidResourceException(api_logical_id, name + " Authorizer must define 'JwtConfiguration'.")
         self.jwt_configuration = jwt_configuration
-        if not id_source:
-            raise InvalidResourceException(api_logical_id, name + " Authorizer must define 'IdentitySource'.")
         self.id_source = id_source
+        self.function_arn = function_arn
+        self.function_invoke_role = function_invoke_role
+        self.identity = identity
+        self.authorizer_payload_format_version = authorizer_payload_format_version
+        self.enable_simple_responses = enable_simple_responses
+
+        # Validate parameters
+        if authorization_scopes is not None and not isinstance(authorization_scopes, list):
+            raise InvalidResourceException(api_logical_id, "AuthorizationScopes must be a list.")
+
+        # Validate necessary parameters exist
+        authorizer_type = self._get_auth_type()
+        if authorizer_type == "JWT":
+            if not self.jwt_configuration:
+                raise InvalidResourceException(
+                    api_logical_id, name + " OAuth2 Authorizer must define 'JwtConfiguration'."
+                )
+            if not self.id_source:
+                raise InvalidResourceException(
+                    api_logical_id, name + " OAuth2 Authorizer must define 'IdentitySource'."
+                )
+
+        if authorizer_type == "REQUEST":
+            if not self.function_arn:
+                raise InvalidResourceException(api_logical_id, name + " Lambda Authorizer must define 'FunctionArn'.")
+            if not self.authorizer_payload_format_version:
+                raise InvalidResourceException(
+                    api_logical_id, name + " Lambda Authorizer must define 'AuthorizerPayloadFormatVersion'."
+                )
+
+    def _get_auth_type(self):
```

Comment:
Now that there is more than one valid type of authorization, I'd prefer to avoid check/default - is any configuration without `jwt_configuration` a valid "REQUEST" authorizer? Or, are there other parameters that need to be present and we should add a third case of "This isn't valid at all, and we're going to fail with a helper message"?

Reply:
There is no assumption that any authorizer is valid by default. We have checks on lines 104-110 to determine if all required parameters for `REQUEST` authorizer are present.

If we add checks mentioned in your previous comment to validate that certain parameters only valid for one auth type - we should cover all scenarios.

Reply:
Exactly right - we just want to make sure we check the different contingent conditions (which I see has changed) and fail with helpful error messages when appropriate.